segmentation_models_pytorch/encoders/timm_mobilenetv3.py

```diff
@@ -65,16 +68,14 @@ def get_stages(self):
                 self.model.blocks[2],
                 self.model.blocks[3:5],
                 self.model.blocks[5:],
-            ]
+            ])
         else:
-            ValueError("MobileNetV3 mode should be small or large, got {}".format(self._mode))
+            raise ValueError("MobileNetV3 mode should be small or large, got {}".format(self._mode))
 
     def forward(self, x):
-        stages = self.get_stages()
-
         features = []
-        for i in range(self._depth + 1):
-            x = stages[i](x)
+        for stage in self._stages:
```

Comment:
If the depth is equal to 4, we do not need to pass the tensor through all 5 stages (this reduces time and space), that is why I iterate over depth

Reply:
Ahh okay, thanks for the feedback. AFAIK there was an issue with looking up `self._stages[i]`, but I'll see if I can work around that.

Reply:
I changed the code to use `_depth` again. It's not my preferred way to code that behavior, but other methods didn't appear to be scriptable. Is this acceptable?